src/main/java/com/github/manolo8/darkbot/utils/login/LoginUtils.java

```diff
@@ -167,12 +187,11 @@ private static String replaceParameters(String params) {
         return params;
     }
 
-    private static String getLoginUrl(InputStream in) {
-        return new BufferedReader(new InputStreamReader(in)).lines()
-                .map(LOGIN_PATTERN::matcher)
-                .filter(Matcher::find)
-                .map(matcher -> matcher.group(1).replace("&amp;", "&"))
-                .findFirst().orElseThrow(WrongCredentialsException::new);
+    private static String getLoginUrl(String in) {
+        Matcher match = LOGIN_PATTERN.matcher(in);
+        if (match.find()) return match.group(1).replace("&amp;", "&");
+
+        throw new LoginException();
```

Comment:
```suggestion
        throw new LoginException("Failed to get login URL in frontpage");
```
Make exceptions descriptive when possible